lib/mrtrix3/run.py

```diff
@@ -335,3 +338,52 @@ def versionMatch(item):
 
   app.error('Unable to find executable for MRtrix3 command ' + item)
 
+
+
+# If the target executable is not a binary, but is actually a script, use the
+#   shebang at the start of the file to alter the subprocess call
+def _shebang(item):
+  import os
+  from mrtrix3 import app
+  from distutils.spawn import find_executable
+  # If a complete path has been provided rather than just a file name, don't perform any additional file search
+  if os.sep in item:
+    path = item
+  else:
+    path = versionMatch(item)
+    if path == item:
+      path = find_executable(exeName(item))
+  if not path:
+    app.debug('File \"' + item + '\": Could not find file to query')
+    return []
+  # Read the first 1024 bytes of the file
+  with open(path, 'rb') as f:
+    data = f.read(1024)
+  # List of permissible text characters
+  textchars = bytearray({7,8,9,10,12,13,27} | set(range(0x20, 0x100)) - {0x7f})
+  # Are there any non-text characters? If so, it's a binary file, so don't go looking for a shebang
```

Comment:
> As long as binaries are guaranteed to have some deterministic code at the start, sure this can go.

I think this is a safe assumption, although the one case that troubles me is static binaries. I'm not sure whether they have a fixed 'magic number' at the start. If they really just consist of raw binary without any form of lead-in, then there is an outside chance that static executables might start with that combination - the chance is only ~1 in 65,000, but...